wmsm-app: Segfault caused by infinite recursion in installPhase

**Describe the bug**
`wmsm-app`, part of windowmaker's `dockapps`, fails to install due to an error in the `preInstall` hook.

```
installing 'wmsm.app-0.2.1'
these derivations will be built:
  /nix/store/d4q3q4hxyd7q2z8czymknpbcjgfs2jx1-wmsm.app-0.2.1.drv
building '/nix/store/d4q3q4hxyd7q2z8czymknpbcjgfs2jx1-wmsm.app-0.2.1.drv'...
unpacking sources
unpacking source archive /nix/store/ycgrvfk9fxvqfjs95vidbmydr5hvi8dl-dockapps-b2b9d87
source root is dockapps-b2b9d87/wmsm.app/wmsm
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/x0dcb2rxlzf32g0ddfkqqz1sfcyx4yay-bash-4.4-p23/bin/bash CC=cc
cc -Wall -O3   -c -o wmsm.o wmsm.c
wmsm.c: In function 'main':
wmsm.c:433:5: warning: ignoring return value of 'fgets' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  433 |     fgets(line,sizeof(line),file);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
wmsm.c: In function 'get_values':
wmsm.c:273:5: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  273 |     fscanf(file,"%f",&uptime);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
cc -Wall -O3 -o wmsm wmsm.o -L/usr/X11R6/lib -lXpm -lXext -lX11 -ldockapp
installing
builder for '/nix/store/d4q3q4hxyd7q2z8czymknpbcjgfs2jx1-wmsm.app-0.2.1.drv' failed due to signal 11 (Segmentation fault)
error: build of '/nix/store/d4q3q4hxyd7q2z8czymknpbcjgfs2jx1-wmsm.app-0.2.1.drv' failed
```

**To Reproduce**

```
nix-env -iA nixos.dockapps.wmsm-app
```

**Additional context**

It's because of infinite recursion in `preInstall`. `preInstall` should be defined as `installPhase` instead.

https://github.com/NixOS/nixpkgs/blob/664ebda061df7791a14aac226ccda5526d218632/pkgs/applications/window-managers/windowmaker/dockapps/wmsm-app.nix#L29-L33

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@bstrik 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.45, NixOS, 21.05.1353.c06613c25df (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.12`
 - channels(sae): `"nixos-21.11pre300903.00c86ad1463"`
 - channels(root): `"nixos-21.05.1353.c06613c25df"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/window-managers/windowmaker/dockapps/wmsm-app.nix
+++ b/pkgs/applications/window-managers/windowmaker/dockapps/wmsm-app.nix
@@ -26,7 +26,7 @@ stdenv.mkDerivation rec {
 
   makeFlags = [ "CC=${stdenv.cc.targetPrefix}cc" ];
 
-  preInstall = ''
+  installPhase = ''
     runHook preInstall
     install -d ${placeholder "out"}/bin
     runHook postInstall

--- a/pkgs/development/idris-modules/build-idris-package.nix
+++ b/pkgs/development/idris-modules/build-idris-package.nix
@@ -36,9 +36,10 @@ stdenv.mkDerivation ({
   # Some packages use the style
   # opts = -i ../../path/to/package
   # rather than the declarative pkgs attribute so we have to rewrite the path.
-  postPatch = ''
+  patchPhase = ''
     runHook prePatch
     sed -i ${ipkgName}.ipkg -e "/^opts/ s|-i \\.\\./|-i ${idris-with-packages}/libs/|g"
+    runHook postPatch
   '';
 
   buildPhase = ''

